File: placementprepration/Nqueens.java
import  java.util.*;
public class Nqueens{
    public static void solve(int n){
        placeQueens(new int [n] ,0);
    }
    static void placeQueens(int []board,int row){
        if(row==board.length){
            printBoard(board);
            return;
        }
        for(int col=0;col<board.length;col++){
            if(issafe(board,row,col)){
                board[row]=col;
                placeQueens(board, row+1);
            }
        }
    }
    static boolean issafe(int []board,int row,int col){
        for(int i=0;i<row;i++){
            if(board[i]==col || Math.abs(board[i]-col)==Math.abs(i-row)){
                return false;
            }
        }
        return true;
    }
    static void printBoard(int []board){
        for(int row:board){
            System.out.println(".".repeat(row)+"Q"+".".repeat(board.length-row-1));
        }
        System.out.println();
    }
    public static void main(String [] args){
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        solve(n);
    }
}

File: Nqueens.py
import sys


def is_safe(board, row, col):
    for r in range(row):
        c = board[r]
        if c == col or abs(c - col) == abs(r - row):
            return False
    return True


def place_queens(board, row, out_lines):
    n = len(board)
    if row == n:
        for r in board:
            out_lines.append("." * r + "Q" + "." * (n - r - 1))
        out_lines.append("")
        return
    for col in range(n):
        if is_safe(board, row, col):
            board[row] = col
            place_queens(board, row + 1, out_lines)


def solve(n):
    board = [-1] * n
    lines = []
    place_queens(board, 0, lines)
    return lines


def main():
    data = sys.stdin.read().strip().split()
    if not data:
        return
    n = int(data[0])
    for line in solve(n):
        print(line)


if __name__ == "__main__":
    main()
